analysis/loader/loader.go

```diff
@@ -0,0 +1,111 @@
+// Package loader wraps golang.org/x/tools/go/packages with utility types and methods
+package loader
+
+// this package mostly exists due to cruft in migrating from
+// golang.org/x/tools/go/loader to golang.org/x/tools/go/packages
+// where certain types and functions were more convenient to put here
+// rather than do significant reworking of existing package APIs (which,
+// looked to worsen APIs and not improve them, thus...). this package
+// relies heavily on the packages package to do the heavy lifting.
+
+import (
+	"go/ast"
+	"go/token"
+	"go/types"
+
+	"golang.org/x/tools/go/ast/astutil"
+	"golang.org/x/tools/go/packages"
+)
+
+// Program provides access to various elements of a type checked set of packages.
+// Fset and AllPackages are not so easily combined, we often need to query the
+// entire program ast which contains all packages, often needing additional
+// information within any given package after that.
+type Program struct {
+	// Fset contains the entire program Fset
+	Fset *token.FileSet
+
+	// AllPackages is the list of all loaded packages for a program.
+	// It is this exposed this way to match the old loader API.
+	AllPackages map[*types.Package]*packages.Package
+}
+
+// Load loads a package, calling packages.Load
+func Load(conf *packages.Config, errorH func(error), args ...string) (*Program, error) {
```

Comment:
All of the tests that were touched here now pass in `conf`, but none of them were updated to pass `errorH`.

Reply:
yea, good eye. the errorH function was basically to handle cgo somewhat more gracefully by catching it and spitting out a legible error. the new modules handles cgo better than the loader program so I believe it's dead code, though the broken test seems to be a bug maybe in modules handling (I gave up at the time). thanks for catching, should prob remove if that's the case